### PR TITLE
fix: bundle spectral versions into rhoas guidelines using peerDepende…

### PR DESCRIPTION
**Pull request by dimakis**
…ncies

### CODE REVIEW

**Comment by dimakis** [comment]
@wtrocki addition of peerDependancies and updated to only allow for increases in patch number

File: spectral/package.json [hunk 1]
```diff
@@ -14,8 +14,8 @@
     "cli-integration-tests": "./scripts/rules.sh"
   },
   "dependencies": {
-    "@stoplight/spectral-cli": "^6.4.1",
-    "@stoplight/spectral-core": "^1.12.3"
+    "@stoplight/spectral-cli": "~6.4.1",
+    "@stoplight/spectral-core": "~1.12.3"
```
Comment (wtrocki) [inline spectral/package.json:18]: What is the point of peerDependencies if they are in dependencies
```suggestion
```

File: spectral/package.json [hunk 2]
```diff
@@ -45,5 +45,9 @@
   "bugs": {
     "url": "https://github.com/redhat-developer/app-services-api-guidelines/issues"
   },
-  "homepage": "https://github.com/redhat-developer/app-services-api-guidelines#readme"
+  "homepage": "https://github.com/redhat-developer/app-services-api-guidelines#readme",
+  "peerDependencies": {
+    "@stoplight/spectral-cli": "~6.4.1",
+    "@stoplight/spectral-core": "~1.12.3"
```
Comment (wtrocki) [inline spectral/package.json:51]:
```suggestion
```

File: spectral/package.json [hunk 3]
```diff
@@ -45,5 +45,9 @@
   "bugs": {
     "url": "https://github.com/redhat-developer/app-services-api-guidelines/issues"
   },
-  "homepage": "https://github.com/redhat-developer/app-services-api-guidelines#readme"
+  "homepage": "https://github.com/redhat-developer/app-services-api-guidelines#readme",
+  "peerDependencies": {
+    "@stoplight/spectral-cli": "~6.4.1",
```
Comment (wtrocki) [inline spectral/package.json:50]:
```suggestion
    "@stoplight/spectral-cli": "6.4.1",
```

File: spectral/package.json (hunk 1 above)
Comment (dimakis) [reply spectral/package.json:18]: am I wrong about these? if the `~` is specified the last number ( for ~1.2.3 -> will be 3, for ~1.2 -> will be 2) will not be a hard requirement. this allows a patch increase to the dependancy which, if semver is properly being followed, should not mean a breaking change but a safer set of dependancies as patches are usually bug fixes right?
https://github.com/npm/node-semver#tilde-ranges-123-12-1

Comment (wtrocki) [reply spectral/package.json:18]: https://docs.renovatebot.com/dependency-pinning/#why-pin-dependencies

Comment (dimakis) [reply spectral/package.json:18]: sold on pinning 😄

**Comment by dimakis** [comment]
> This is good but having dependencies in both peer and depenency is redundant. We can merge once that is sorted.

fails to build unless dependancies are present. you must do both it seems. so are you saying take `peerDependancies` out? that was the method that the previous release was at anyway?

File: spectral/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@rhoas/spectral-ruleset",
-  "version": "0.1.4",
+  "version": "0.2.0dev",
```
Comment (wtrocki) [inline spectral/package.json:3]:
```suggestion
  "version": "0.2.0-dev",
```

**Review by wtrocki — APPROVED** [review]

**Pull request merged**